lib/collection/src/shards/replica_set/execute_read_operation.rs

```diff
@@ -122,6 +122,8 @@ impl ShardReplicaSet {
             )));
         };
 
+        self.check_partial_snapshot_read_lock()?;
```

Comment:
I could not reproduce the problem here, but It feels like we should better check the lock before trying to real local shard.

Reply:
nevermind, I see now that we actually want to check only after real lock is aquired.